app/src/main/java/com/github/se/wanderpals/model/repository/TripsRepository.kt

```diff
@@ -536,58 +563,63 @@ class TripsRepository(
       withContext(dispatcher) {
         Log.d("TripsRepository", "addTripId: Adding tripId to user")
 
-        val userDocumentRef = usersCollection.document(uid)
-
-        // Ensure the user's document exists before attempting to modify it.
-        // If the document does not exist, create it with an initial empty list of tripIds.
-        val userDoc = userDocumentRef.get().await()
-        if (!userDoc.exists()) {
-          // Initialize the document with an empty tripIds list.
-          userDocumentRef.set(mapOf("tripIds" to listOf<String>())).await()
-        }
+        if (!isTripIdValid(tripId)) {
+          false
+        } else {
 
-        try {
-          val transactionResult =
-              firestore
-                  .runTransaction { transaction ->
-                    val snapshot = transaction.get(userDocumentRef)
+          val userDocumentRef = usersCollection.document(uid)
 
-                    // Safely attempt to retrieve and cast the tripIds list from the snapshot.
-                    val existingTripIds: MutableList<String> = mutableListOf()
+          // Ensure the user's document exists before attempting to modify it.
+          // If the document does not exist, create it with an initial empty list of tripIds.
+          val userDoc = userDocumentRef.get().await()
+          if (!userDoc.exists()) {
+            // Initialize the document with an empty tripIds list.
+            userDocumentRef.set(mapOf("tripIds" to listOf<String>())).await()
+          }
 
-                    val rawTripIds = snapshot["tripIds"]
-                    if (rawTripIds is List<*>) {
-                      // Filter non-null and String values only, safely adding them to
-                      // existingTripIds.
-                      existingTripIds.addAll(rawTripIds.filterIsInstance<String>())
+          try {
```

Comment:
Cognitive Complexity is seen as a problem with Sonar but the fact that your code has great variable names and great structure described with comments renders this issues not needing a fix from my end